README.md

```diff
@@ -245,6 +245,131 @@ in the correct format as Protobuf encoded binary data.
 `ProtobufDescriptor` is preferred as you might have to pay a performance penalty
 should you want to use the `BigQuerySchema` instead.
 
+### Batch client
+
+The batch client can be used if you want to upload a big dataset of data to bigquery without any additional cost.
+
+Here follows an example on how you can create such a batch API driven BigQuery client.
+
+```go
+import (
+    "bytes"
+    "context"
+    "encoding/json"
+
+    "github.com/OTA-Insight/bqwriter"
+
+    "cloud.google.com/go/bigquery"
+)
+
+// TODO: use more specific context
+ctx := context.Background()
+
+// create a BQ (stream) writer thread-safe client,
+bqWriter, err := bqwriter.NewStreamer(
+    ctx,
+    "my-gcloud-project",
+    "my-bq-dataset",
+    "my-bq-table",
+    &bqwriter.StreamerConfig{
+        // use 5 background worker threads
+        WorkerCount: 5,
+        // create the client with the JSON format and using autodetect
+        BatchClient: &bqwriter.BatchClientConfig{
+            AutoDetect: true,
```

Comment:
I would add a comment here on what it is auto-detecting,
while you might (and should) have documented that in the config, it's best that an example is also immediately obvious in its intention, be its not it internals.